components/src/BarcodeScanner/barcodescanner.tsx

```diff
@@ -0,0 +1,187 @@
+/**
+ * Copyright © 2019 Elastic Path Software Inc. All rights reserved.
+ *
+ * This is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this license. If not, see
+ *
+ *     https://www.gnu.org/licenses/
+ *
+ *
+ */
+
+import React from 'react';
+import Quagga from 'quagga/dist/quagga.js';
+import Modal from 'react-responsive-modal';
+import { getConfig } from '../utils/ConfigProvider';
+import './barcodescanner.less';
+
+let intl = { get: str => str };
+
+interface BarcodeScannerProps {
+  handleModalClose: (...args: any[]) => any,
+  handleCodeFound: (...args: any[]) => any,
+  isModalOpen: boolean,
+}
+interface BarcodeScannerState {
+  barcodes: any[]
+}
+
+export default class BarcodeScanner extends React.Component<BarcodeScannerProps, BarcodeScannerState> {
+  constructor(props) {
+    super(props);
+    this.state = {
+      barcodes: [],
+    };
+    const epConfig = getConfig();
+    ({ intl } = epConfig);
+    this.onDetected = this.onDetected.bind(this);
+    this.onModalClose = this.onModalClose.bind(this);
+    this.initScanner = this.initScanner.bind(this);
+    this.stopScanning = this.stopScanning.bind(this);
+    this.scannerContainer = React.createRef();
+  }
+
+  componentDidMount(): void {
+    this.initScanner();
+  }
+
+  componentWillUnmount(): void {
+    this.stopScanning();
+  }
+
+  static checkAvailability(): boolean {
+    return Boolean(navigator && navigator.mediaDevices);
+  }
+
+  initScanner() {
+    const scannerContainer = this.scannerContainer.current;
+    // eslint-disable-next-line react/destructuring-assignment
+    console.warn(this.state.barcodes);
+    Quagga.init({
+      inputStream: {
+        name: 'Live',
+        type: 'LiveStream',
+        target: scannerContainer,
+        constraints: {
+          width: 480,
+          height: 320,
+          facingMode: 'environment',
+        },
+      },
+      numOfWorkers: navigator.hardwareConcurrency,
+      decoder: {
+        readers: [
+          'code_128_reader',
+          'ean_reader',
+          'code_39_reader',
+          'code_39_vin_reader',
+          'codabar_reader',
+          'upc_reader',
```

Comment:
I think you'll just need to keep `upc_reader ` for the implementation. The others may confuse the reader.